historical_surveillance/views.py

```diff
@@ -0,0 +1,328 @@
+            # change the updated by column to automatically insert the name of the logged in user updating the column
+            district_updated = form.cleaned_data['updated_by']
+            if not District.objects.filter(District_Code=district_code,
+                                         created_by=district_created_by,
+                                         updated_by=district_updated,
+                                         District_Name=district_name).exists():
+                form.save()
+                return redirect("/")
+
+            else:
+                error_message = "The record you are trying to create already exists."
+
+        stu = {
+
+            "error_message": error_message,
+
+        }
+        return render(request, 'district.html', stu)
+    return render(request, 'district.html', {'form': form, 'district_created': data})
+
+
+# This function controls the schools creation form view and process form data for save into the database
+def create_school(request):
+    error_message = None
+    form = SchoolForms()
+    data = School.objects.all()
+    if request.method == 'POST':
+        form = SchoolForms(request.POST)
+        if form.is_valid():
+            # change the created by column to automatically insert the logged in user
+            school_created_by = form.cleaned_data['created_by']
+            school_code = form.cleaned_data['School_Code']
+            school_name = form.cleaned_data['School_Name']
+            district_name = form.cleaned_data['District_Name']
+            school_category = form.cleaned_data['category_of_school']
+            # change the updated by column to automatically insert the name of the logged in user updating the column
+            school_updated = form.cleaned_data['updated_by']
+            if not School.objects.filter(School_Code=school_code,
+                                         created_by=school_created_by,
+                                         updated_by=school_updated,
+                                         School_Name=school_name,
+                                         District_Name=district_name,
+                                         category_of_school=school_category).exists():
+                form.save()
+                return redirect("/")
+
+            else:
+                error_message = "The record you are trying to create already exists."
+
+        stu = {
+            "name_of_school": data,
+            "error_message": error_message,
+            "form" : form,
+        }
+        return render(request, 'school.html', stu)
+    return render(request, 'school.html', {'form':form, 'name_of_school':data})
+
+
+# This function controls the enrollment by school,
+# grade and gender form view, processes the and submits form data to the database
+def enrollment(request):
+    form = EnrollmentForms()
+    district_df = pd.DataFrame(District.objects.all().values())
+    districts_names = district_df['District_Name']
```

Comment:
If the goal here is to just get the district names as a list we can just do

This will return a map: [{district_name: 'distcit 1' }, {district_name: district2}, ...]
`District.objects.values('district_name').distinct()`

But we can get a list of unique district names by doing:
`District.objects.values_list('district_name', flat=True).distinct()`

Will return a unique list.

No data frame needed and we don't query all the data to memory.